src/azure.ad/types_error.bal

```diff
@@ -14,17 +14,15 @@
 // specific language governing permissions and limitations
 // under the License.
 
-public const AD_CLIENT_ERROR = "{ballerinax/azure.ad}AdClientError";
-
-public type AdClientError error<string, Details>;
+public type AdClientError error;
```

Comment:
To make the error a nominal type that works within Ballerina's structural type system, we should make the error `distinct`. Then user can perform type test to distinguish these errors from the generic error

```suggestion
public type AdClientError distinct error;
```

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerinax-azure.ad/pull/5/commits/22d494bd94ccc89dd2912d76f3e26db93cc75af7